### ISSUE TITLE
If tailscale is "Connected" you won't be able to connect to public wifi at SJC, TUC, and other airports

### ISSUE THREAD

**Issue by skirsch** [tailscale/tailscale#11553]
### What is the issue?

I noticed that I couldn't connect to wifi public airport networks after installing tailscale.

turns out if tailscale is in CONNECTED state, you won't be able to connect at SJC, TUC, etc. airports.

Switching tailscale to DISCONNECTED state enabled me to connect to the airport wifi networks.

I have no idea why just certain airports, but this is 100% repeatable.

### Steps to reproduce

connect to tailscale. 
close laptop
open laptop at SJC or TUC.
You will connect to the wifi, but there will be No Internet.

set tailscale to disconnected and retry to connect and bingo.

I'm on a Windows 11 lenovo laptop.

### OS

Windows

### OS version

11

### Tailscale version

1.62.1

### Bug report

BUG-6b96a26101df7d8442d1be0f7df5571c592df96c25857cba4451d12144f964f4-20240328222619Z-02d8b1a0c7d119be

**Comment by wingcomm** [comment]
This looks like a duplicate of https://github.com/tailscale/tailscale/issues/1634

**Comment by MannanSaood** [comment]
This is sometimes possible as Public Wifi has blocked VPN. It can be solved using OpenVPN protocol but tailscale is contrary to OpenVPN as tailscale uses P2P mesh but OpenVPN uses concentrator that funnels traffic between devices.

**Comment by skirsch** [comment]
Yes and it still isn’t fixed!!!

I had exactly the same experiences that person this week. My phone Tailscale worked fine to connect, but my laptop did not.


Sent from my iPhone

On Mar 28, 2024, at 11:56 PM, wingcomm ***@***.***> wrote:

﻿

This looks like a duplicate of #1634<https://github.com/tailscale/tailscale/issues/1634>

**Comment by skirsch** [comment]
I'm at the Taipei airport. The main airport wifi connected flawlessly, the wifi in Cathay Pacific lounge required me to disable tailscale.

The average user is going to scratch their head and never figure out that they left tailscale enabled.

**Comment by ericswpark** [comment]
Same issue here at Chicago O'Hare (ORD). When Tailscale is running on my NixOS laptop, Internet connectivity does not work at all and the captive portal page for the Boingo hotspot doesn't show up. Surprisingly, Tailscale Android doesn't seem to suffer from the same problem.

Disabling Tailscale allows Internet connectivity and Firefox immediately recognizes the captive portal, but when I reactivate Tailscale after clicking through the "free WiFi" steps on the captive portal the Internet connectivity drops again.

Perhaps the IP range assigned by the public WiFi conflicts with Tailscale's? ORD was using `100.72.177.0/16` subnet `100.72.255.255`.

**Comment by MannanSaood** [comment]
@ericswpark  i have noticed that many routers can block DNS requests as many routers in airports are using DNS filtering systems, as the 'Magic DNS' is not a very common DNS, the cause might be it is getting blocked by that filter, so once try disabling is onboard DNS. This issue occurred when I changed my router as well, I had to remove the DNS Filter to use the internet.

**Comment by bcreane** [comment]
Hi @skirsch,

Thanks for the report. Sounds like potential friction between Tailscale and captive portals (like those at SJC, TUC, and ORD).

To help us pinpoint whether this is a DNS blocking issue or a subnet IP conflict, could you try the following debugging steps next time you encounter this?

**1. DNS & Captive Portal Checks (Most Likely Cause)**
Captive portals often use DNS redirection to force the login page to appear. Tailscale's DNS settings might be interfering with this interception.

* **Disable Magic DNS:** Try temporarily disabling "Magic DNS" in your Tailscale Admin Console.
* **HTTP Trigger:** Try navigating to `http://neverssl.com` or `http://8.8.8.8` in your browser while connected. This forces a non-secure connection, which often triggers the portal more reliably than HTTPS pages.

**2. IP Subnet Conflicts**
As @ericswpark noted, some public networks (like ORD) assign IPs in the `100.64.0.0/10` Carrier Grade NAT range, which overlaps with Tailscale's own IP space.

* **Check Local IP:** When connected to the airport Wi-Fi (even without internet), check the IP address assigned to your physical Wi-Fi adapter (run `ipconfig` in Command Prompt). If it starts with `100.x.x.x`, it is likely conflicting with Tailscale.

**3. Exit Nodes**

* If you have an **Exit Node** enabled, try disabling it. Tunneling all traffic (including local network discovery traffic) can sometimes prevent the captive portal from loading.

**Current Workaround**
For now, the consistent workaround remains:

1. **Disconnect** Tailscale.
2. Connect to the Wi-Fi and complete the login/portal.
3. **Reconnect** Tailscale once internet access is verified.

If you are able to reproduce this state again, please generate a bug report **while the issue is happening** (Right-click Tailscale icon > Bug Report) and share the ID here. This will let us see the specific routing table clash.

**Comment by skirsch** [comment]
THanks. will give that a shot next time I'm in the airport.

**Comment by future-figs** [comment]
Just hit this issue at LGA. What a pain! Took me an hour to figure out what was wrong.

1. `http://neverssl.com` and `http://8.8.8.8` both time out. In fact, everything timed out except for `nslookup` commands with the DNS server explicitly specified (1.1.1.1 or 8.8.8.8). Looking up the hotspot portal's domain and punching that IP into a web browser yields...timeout.
2. As a matter of fact, my laptop has been assigned a `100.x.y.z` IP address.
3. I have on exit node configured but not in active use, so none of my regular traffic should be routed to it.

Once connected, re-enabling Tailscale immediately causes the issue to return, but I found a workaround: manually add DNS servers for the hotspot wifi network. This suggests that all DNS traffic is getting indiscriminately sent to Tailscale instead of the local network's automatic DNS servers. I didn't experiment with disabling MagicDNS or anything.

**Comment by skirsch** [comment]
There is another issue.... If you have tailscale running, then use expressVPN connect and then disconnect, tailscale magic DNS is non functional until you disconnect and reconnect. 

Ugh!
